tests/e2e/specs/backend/product-query/advanced-filters.test.ts

```diff
@@ -0,0 +1,160 @@
+/**
+ * External dependencies
+ */
+import {
+	saveOrPublish,
+	selectBlockByName,
+	findToolsPanelWithTitle,
+	getFixtureProductsData,
+	shopper,
+	getToggleIdByLabel,
+} from '@woocommerce/blocks-test-utils';
+import { ElementHandle } from 'puppeteer';
+import {
+	setCheckbox,
+	unsetCheckbox,
+	ensureSidebarOpened,
+} from '@woocommerce/e2e-utils';
+
+/**
+ * Internal dependencies
+ */
+import { GUTENBERG_EDITOR_CONTEXT, describeOrSkip } from '../../../utils';
+import {
+	block,
+	SELECTORS,
+	resetProductQueryBlockPage,
+	toggleAdvancedFilter,
+	getPreviewProducts,
+	getFrontEndProducts,
+	clearSelectedTokens,
+	selectToken,
+} from './common';
+
+describeOrSkip( GUTENBERG_EDITOR_CONTEXT === 'gutenberg' )(
+	`${ block.name } > Advanced Filters`,
+	() => {
+		let $productFiltersPanel: ElementHandle< Node >;
+		const defaultCount = getFixtureProductsData().length;
+		const saleCount = getFixtureProductsData( 'sale_price' ).length;
+		const outOfStockCount = getFixtureProductsData( 'stock_status' ).filter(
+			( status: string ) => status === 'outofstock'
+		).length;
+
+		beforeEach( async () => {
+			/**
+			 * Reset the block page before each test to ensure the block is
+			 * inserted in a known state. This is also needed to ensure each
+			 * test can be run individually.
+			 */
+			await resetProductQueryBlockPage();
+			await ensureSidebarOpened();
+			await selectBlockByName( block.slug );
+			$productFiltersPanel = await findToolsPanelWithTitle(
+				'Advanced Filters'
+			);
+		} );
+
+		/**
+		 * Reset the content of Product Query Block page after this test suite
+		 * to avoid breaking other tests.
+		 */
+		afterAll( async () => {
+			await resetProductQueryBlockPage();
+		} );
+
+		describe( 'Sale Status', () => {
+			it( 'Sale status is disabled by default', async () => {
+				await expect( $productFiltersPanel ).not.toMatch(
+					'Show only products on sale'
+				);
+			} );
+
+			it( 'Can add and remove Sale Status filter', async () => {
+				await toggleAdvancedFilter( 'Sale status' );
+				await expect( $productFiltersPanel ).toMatch(
+					'Show only products on sale'
+				);
+				await toggleAdvancedFilter( 'Sale status' );
+				await expect( $productFiltersPanel ).not.toMatch(
+					'Show only products on sale'
+				);
+			} );
+
+			it( 'Editor preview shows correct products corresponding to the value `Show only products on sale`', async () => {
+				expect( await getPreviewProducts() ).toHaveLength(
+					defaultCount
+				);
+				await toggleAdvancedFilter( 'Sale status' );
+				await setCheckbox(
+					await getToggleIdByLabel( 'Show only products on sale' )
+				);
+				expect( await getPreviewProducts() ).toHaveLength( saleCount );
+				await unsetCheckbox(
+					await getToggleIdByLabel( 'Show only products on sale' )
+				);
+				expect( await getPreviewProducts() ).toHaveLength(
+					defaultCount
+				);
```

Comment:
This is a great direction, thank you so much, I ended up structuring tests like following:
```
- Enable Sale Status > Editor preview shows only on sale products
- Enable Sale Status > On the front end, block shows only on sale products
```

I moved the tests for the default state to the main `describe` block to avoid duplicated tests. Same for your review about stock status.

Some notes:
- I use `>` in the test name instead of creating another nested `describe` block to save one or two indentation levels, too many nested `describe` blocks make the test harder to read.
- I explicitly mention the expected behavior in the test title.